tests/integration/dab_rbac/test_crud_permissions.py

```diff
@@ -37,56 +39,54 @@ def get_detail_url(obj, skip_if_not_found=False):
         raise
 
 
-@pytest.mark.django_db
-@pytest.mark.parametrize("model", permission_registry.all_registered_models)
-def test_factory_sanity(model, cls_factory):
-    cls_factory.create(model)
-
-
 @override_settings(DEBUG=True)
 @pytest.mark.django_db
 @pytest.mark.parametrize("model", permission_registry.all_registered_models)
 def test_add_permissions(
+    request,
     model,
     cls_factory,
-    user,
-    user_api_client,
+    default_user,
+    user_client,
     give_obj_perm,
-    preseed_credential_types,
 ):
-    create_data = cls_factory.get_create_data(model)
-    data = cls_factory.get_post_data(model, create_data)
+    model_name = cls_factory.get_model_name(model)
+    model_obj = cls_factory.get_fixture_object(request, model_name)
+    post_data = cls_factory.get_post_data(model_obj)
+
     if "add" not in model._meta.default_permissions:
         pytest.skip("Model has no add permission")
 
     url = reverse(f"{get_basename(model)}-list")
 
-    response = user_api_client.post(url, data=data)
+    response = user_client.post(url, data=post_data)
     prior_ct = model.objects.count()
     assert response.status_code == 403, response.data
     assert model.objects.count() == prior_ct  # assure nothing was created
 
     # Figure out the parent object if we can
     parent_field_name = permission_registry.get_parent_fd_name(model)
     if parent_field_name:
-        parent_obj = create_data[parent_field_name]
+        parent_obj = permission_registry.get_parent_model(model).objects.get(
+            id=post_data[parent_field_name]
+        )
```

Comment:
It might be more direct to do `getattr(model_obj, parent_field_name)`, although this assumes parity between `post_data` and the reference object, I think we have that. Either way, just minor suggestion.